### PR TITLE
SOC-478 Validation Errors for Facebook Reg

### PR DESCRIPTION
**Pull request by bkoval**
This adds validation errors on Facebook Registration page. Work in Progress as we have Services ticket blocking this one and we need to make sure we cover all the cases, but please feel free to comment on this code here.

@kvas-damian @rafalkalinski

### CODE REVIEW

File: front/scripts/auth/facebook/FacebookRegistration.ts [hunk 1]
```diff
@@ -94,15 +96,15 @@ class FacebookRegistration {
 			if (status === HttpCodes.OK) {
 				window.location.href = this.redirect;
 			} else if (status === HttpCodes.BAD_REQUEST) {
-				//ToDo show the "unable to login" error
+                this.formErrors.displayGeneralError();
```
Comment (kvas-damian) [inline front/scripts/auth/facebook/FacebookRegistration.ts]: tabs vs spaces

File: front/scripts/auth/common/FormErrors.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
+class FormErrors {
+	form: HTMLFormElement;
+	generalValidationErrors: Array<string> = ['email_blocked', 'username_blocked', 'birthdate_below_min_age'];
```
Comment (kvas-damian) [inline front/scripts/auth/common/FormErrors.ts]: Please remove this variable from ```SignupForm.ts```

File: front/scripts/auth/common/FormErrors.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+	generalErrorShown: boolean = false;
```
Comment (kvas-damian) [inline front/scripts/auth/common/FormErrors.ts]: The same for this one.

File: front/scripts/auth/common/FormErrors.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+	public translateValidationError(errCode: string): string {
+		return i18n.t('errors.' + errCode);
+	}
+
+	public trackValidationErrors(errors: Array<string>): void {
+		M.track({
+			trackingMethod: 'both',
+			action: M.trackActions.error,
+			category: 'user-login-mobile',
```
Comment (kvas-damian) [inline front/scripts/auth/common/FormErrors.ts]: Please separate tracking for email and facebook registration

Comment (bkoval) [inline front/scripts/auth/common/FormErrors.ts]: Addressed, constructor takes a label prefix.

File: front/scripts/auth/common/FormErrors.ts (hunk 3 above)
Comment (bkoval) [inline front/scripts/auth/common/FormErrors.ts]: Done.

File: front/scripts/auth/common/FormErrors.ts (hunk 2 above)
Comment (bkoval) [inline front/scripts/auth/common/FormErrors.ts]: Done.

File: front/scripts/auth/facebook/FacebookRegistration.ts [hunk 5]
```diff
@@ -26,6 +26,7 @@ class FacebookRegistration {
 	redirect: string;
 	urlHelper: UrlHelper;
 	marketingOptIn: MarketingOptIn;
+    formErrors: FormErrors;
```
Comment (kvas-damian) [inline front/scripts/auth/facebook/FacebookRegistration.ts]: tabs vs spaces?

**Pull request closed without merging**